### ISSUE TITLE
AppNotificationService crashes on packaged application

### ISSUE THREAD

**Issue by gmgallo** [microsoft/TemplateStudio#4596]
### Describe the bug

When the app is packaged. The application crashes on startup. The offending line is this in the Initialize() method of the service is this:
`      AppNotificationManager.Default.Register();`
Looking at the[ AppNotificationManager.Register Method](https://docs.microsoft.com/en-us/windows/windows-app-sdk/api/winrt/microsoft.windows.appnotifications.appnotificationmanager.register?view=windows-app-sdk-1.1#remarks), the Remarks section says: 
 _"For packaged apps, the COM server is defined in the app manifest. The process calling Register and the process defined in the manifest as the COM server are required to be the same."_
So it seems that the service is not mentioned as the COM server in the manifest.

### To Reproduce
Create a WinUI app with Template Studio including AppNotificationService and MSIX package. Run unpacked should work fine. Run the packaged version crashes.

### Applies to the following platforms:

- [X] WinUI
- [ ] WPF
- [ ] UWP

### About your setup

 - Visual Studio Version:  2022
 - Template Studio Wizard Version: 5.2.1
 - Windows Build: 19042.1889

**Comment by mikebattista** [comment]
I can't repro. Works fine for me out-of-the-box.

What's the call stack you get?

**Comment by gmgallo** [comment]
You mean it works MSIX packaged?
Here is the copy of the exception:
`System.Runtime.InteropServices.COMException
  HResult=0x80004005
  Message=Error HRESULT E_FAIL has been returned from a call to a COM component.
  Source=WinRT.Runtime
  StackTrace:
   at WinRT.ExceptionHelpers.<ThrowExceptionForHR>g__Throw|20_0(Int32 hr)
   at ABI.Microsoft.Windows.AppNotifications.IAppNotificationManagerMethods.Register(IObjectReference _obj)
   at Microsoft.Windows.AppNotifications.AppNotificationManager.Register()
   at SI100DataLogger.Notifications.AppNotificationService.Initialize() in ...\repos\SI100Apps\SI100DataLogger\Services\AppNotificationService.cs:line 29
   at SI100DataLogger.App..ctor() in ...\repos\SI100Apps\SI100DataLogger\App.xaml.cs:line 99
   at SI100DataLogger.Program.<>c.<Main>b__1_0(ApplicationInitializationCallbackParams p) in ...\repos\SI100Apps\SI100DataLogger\obj\x86\Debug\net6.0-windows10.0.19041.0\win10-x86\App.g.i.cs:line 34
   at ABI.Microsoft.UI.Xaml.ApplicationInitializationCallback.Do_Abi_Invoke(IntPtr thisPtr, IntPtr p)
`

I was related to [this page](https://docs.microsoft.com/en-us/windows/apps/windows-app-sdk/notifications/app-notifications/app-notifications-quickstart?tabs=cs) for the solution. It requires a manual edit of package.appxmanifest, which I did, but still doesn't work. Here is the relevant section of my editted package.appxmanifest file:

```
Extensions>
              <!--Specify which CLSID to activate when notification is clicked-->   
              <desktop:Extension Category="windows.toastNotificationActivation">
                <desktop:ToastNotificationActivation ToastActivatorCLSID="7b52b86b-07d6-439e-b4eb-37f44c834cd9" />
              </desktop:Extension>
               <!--Register COM CLSID-->    
              <com:Extension Category="windows.comServer">
                <com:ComServer>
                  <com:ExeServer Executable="$targetnametoken$.exe" DisplayName="G2-SI100 Data Logger" Arguments="----AppNotificationActivated:">
                    <com:Class Id="7b52b86b-07d6-439e-b4eb-37f44c834cd9" />
                  </com:ExeServer>
                </com:ComServer>
              </com:Extension>
      </Extensions>
```

ToastActivatorCLSID is set to the GUID Iidentity Name listed above in the same file

**Comment by mikebattista** [comment]
We already include the COM server in the manifest. Did you add a second one? Also have you tried by setting the Executable name directly without the parameter as shown in the docs and in the generated code?

**Comment by gmgallo** [comment]
You are correct, the problem is that I copied the Package.appxmanifest from an earlier project so I didn't have to recreate all the gazilion image icons and other settings. The problem is that this earlier file didn't have the notification service included. Sorry for the false positive, I saw that this can be done in one of the packaging documentation pages to eliminate the a separate packaging project. Sorry for the false positive.
